docs/design/node_agent_config-design.md

```diff
@@ -0,0 +1,91 @@
+# NodeAgentConfig configuration with restic/kopia
+Date: 2023-07-25
+
+## Abstract
+
+New configuration for the file system backup & restore that will be used by the OADP and allow to choose Restic or Kopia uploader type.
+
+## Background
+
+For the file system backup and restore the Velero may use Restic or Kopia as the uploader mechanism. A number of tests were [performed](https://velero.io/docs/main/performance-guidance/) by the Velero community to compare those mechanisms. In many cases Kopia uploader is a much better performing mechanism and as such should be added to the OADP operator.
+
+## Goals
+
+- A new option `nodeagentconfig` to allow configuration of restic or kopia uploader
+- Backwards compatibility with the current OADP configuration schema options, namely `restic`
+- Preparation for the future deprecation of the current `restic` configuration option (from OADP 1.4+)
+- Allow new schema to be used by the datamover node agent
+
+## Non Goals
+- Enablment of datamover node agent
+- Deprecation of the "restic" configuration option
+- E2E tests for the `kopia` or `restic` uplaoder
+
+## High-Level Design
+
+Since new `nodeagentconfig` configuration option is a sibling of the `restic` one, the `restic` will be used as part of the new `nodeagentconfig` with addition of one `UploaderType` option that will be either `kopia` or `restic`.
+
+Without that option a default `restic` will be used. That default option may change in the future.
+
+## Detailed Design
+
+A new structure will be added, that includes the ResticConfig inline and extends this with one new parameter `UploaderType`:
+
+```go
+type NodeAgentConfig struct {
+	// Embedding ResticConfig
+	// +kubebuilder:validation:Optional
+	ResticConfig `json:",inline"`
```

Comment:
This is slightly different because we use that inline it's in the format `json:",inline"` meaning that *all* fields from ResticConfig will be included here on the same level, so we end up with:

```yaml
                    nodeagentconfig:
                      description: NodeAgent is needed to allow selection between kopia or restic
                      properties:
                        enable:
                          description: enable defines a boolean pointer whether we want the daemonset to exist or not
                          type: boolean
                        podConfig:
                          description: Pod specific configuration
                          properties:
                            env:
                              description: env defines the list of environment variables to be supplied to podSpec
                              items:
                                description: EnvVar represents an environment variable present in a Container.
                                properties:
                                  name:
                                    description: Name of the environment variable. Must be a C_IDENTIFIER.
                                    type: string
                                  value:
[...]
```

Reply:
Cool, @mpryc